cli/spec.py

```diff
@@ -0,0 +1,260 @@
+import ConfigParser
+import os
+import yaml
+import yamlordereddictloader
+
+import clg
+from cli import exceptions
+from cli import utils
+
+SPEC_EXTENSION = '.spec'
+LOOKUP_DICT = {
+    '__DEFAULT__': "default"
+}
+
+TRIM_PARAMS = ['default', 'required', 'requires_only']
+
+
+def cfg_file_to_dict(value):
+    """
+    The custom type for clg spec
+    :param value: the argument value.
+    :return: dict based on a provided ini file.
+    """
+    config = ConfigParser.ConfigParser()
+    with open(value) as fd:
+        config.readfp(fd)
+
+    res_dict = {}
+    for section in config.sections():
+        res_dict[section] = {}
+        for option, value in config.items(section):
+            value = utils.string_to_list(value, append_to_list=False)
+            res_dict[section][option] = value
+
+        res_dict[section].pop('__name__', None)
+
+    return res_dict
+
+
+def parse_args(module_name, config, args=None):
```

Comment:
This function is very long... Try to move the code blocks inside into own functions with meaningful names so it will be easier to read and understand